### DISCUSSION TITLE
Splash screen doesn't close after win32App run

### DISCUSSION THREAD

**Discussion by EvgenPervenenko** [microsoft/WindowsAppSDK#3703] [General]
Good time of day, I have test win32 application with manifest TestApp.exe.manifest

`<msix xmlns="urn:schemas-microsoft-com:msix.v1" publisher="CN=TestPublisher" packageName="TestApp" applicationId="TestApp"/>`

and I am registering AppxManifest.xml for shareTarget using TestApp

```
<Applications>
    <Application Id="TestApp" Executable="TestApp.exe" uap10:TrustLevel="mediumIL" uap10:RuntimeBehavior="win32App" uap10:Parameters="ShareFiles">
      <uap:VisualElements AppListEntry="none" DisplayName="TestApp" Description="TestApp" BackgroundColor="transparent" Square150x150Logo="Assets\TestApp.png" Square44x44Logo="Assets\TestApp.png">
        <uap:SplashScreen Image="Assets\TestApp.png" uap5:Optional="true"/>
      </uap:VisualElements>
      <Extensions>
        <uap:Extension Category="windows.shareTarget">
          <uap:ShareTarget Description="Send to TestApp">
            <uap:SupportedFileTypes>
              <uap:FileType>.jpg</uap:FileType>
              <uap:FileType>.png</uap:FileType>
              <uap:FileType>.gif</uap:FileType>
            </uap:SupportedFileTypes>
            <uap:DataFormat>StorageItems</uap:DataFormat>
            <uap:DataFormat>Bitmap</uap:DataFormat>
          </uap:ShareTarget>
        </uap:Extension>
      </Extensions>
    </Application>
  </Applications>
```
  
  but when I share .png using shareTarget, TestApp.exe run and splash screen not closing, if Executable is wrong it is closing, seems I am missing something

**Comment by lhak** [comment]
Make sure to call ReportCompleted() on the ShareOperation in the activation arguments

**Reply by EvgenPervenenko** [reply]
@lhak thank you for your reply, is it possible to call it through winapi(c++), maybe is there any example

**Reply by lhak** [reply]
I do not have any examples, but here is the api documentation (combination of AppSDK and Windows API):

- Get app instance: https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appinstance.getcurrent?view=windows-app-sdk-1.3

- Get activation args: https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appinstance.getactivatedeventargs?view=windows-app-sdk-1.3

- Check that the Kind property is ExtendedActivationKind.ShareTarget
- Get https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.activation.sharetargetactivatedeventargs?view=winrt-22621 from the Data property

- Get the ShareOperation: https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.activation.sharetargetactivatedeventargs.shareoperation?view=winrt-22621

- Get the files and access to the completion methods: https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.datatransfer.sharetarget.shareoperation?view=winrt-22621

**Reply by EvgenPervenenko** [reply]
@lhak thank you
